### PR TITLE
GEODE-9169: remove netty context switch

### PR DESCRIPTION
**Pull request by nonbinaryprogrammer**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Review by ringles — APPROVED** [review]
Seems good!

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/netty/NettyRedisServer.java [hunk 1]
```diff
@@ -176,8 +176,8 @@ public void initChannel(SocketChannel socketChannel) {
         pipeline.addLast(new WriteTimeoutHandler(10));
         pipeline.addLast(ExecutionHandlerContext.class.getSimpleName(),
             new ExecutionHandlerContext(socketChannel, regionProvider, pubsub,
-                allowUnsupportedSupplier, shutdownInvoker, redisStats, backgroundExecutor,
-                redisPasswordBytes, getPort(), member));
+                allowUnsupportedSupplier, shutdownInvoker, redisStats, redisPasswordBytes,
```
Comment (jdeppe-pivotal) [inline geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/netty/NettyRedisServer.java:179]: The `backgroundExecutor` removed here should also be removed from the constructor and can then also be removed from `GeodeRedisServer`.

**Review by jdeppe-pivotal — CHANGES_REQUESTED** [review]

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/netty/NettyRedisServer.java (hunk 1 above)
Comment (nonbinaryprogrammer) [reply geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/netty/NettyRedisServer.java:175]: thanks!

**Review by DonalEvans — APPROVED** [review]

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisServer.java [hunk 2]
```diff
@@ -94,7 +94,7 @@ public GeodeRedisServer(String bindAddress, int port, InternalCache cache) {
     nettyRedisServer = new NettyRedisServer(() -> cache.getInternalDistributedSystem().getConfig(),
         regionProvider, pubSub,
         this::allowUnsupportedCommands, this::shutdown, port, bindAddress, redisStats,
-        redisCommandExecutor, member);
+        member);
```
Comment (dschneider-pivotal) [inline geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisServer.java:97]: shouldn't the redisCommandExecutor field be removed from this class?

Comment (jdeppe-pivotal) [inline geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisServer.java:97]: This executor can be completely removed from the class now.

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/netty/ExecutionHandlerContext.java [hunk 3]
```diff
@@ -257,8 +204,6 @@ public void channelInactive(ChannelHandlerContext ctx) {
         logger.debug("GeodeRedisServer-Connection closing with " + ctx.channel().remoteAddress());
       }
 
-      commandQueue.clear();
-      commandQueue.offer(TERMINATE_COMMAND);
```
Comment (dschneider-pivotal) [inline geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/netty/ExecutionHandlerContext.java:261]: It looks like TERMINATE_COMMAND is no longer used. This class creates in a private static final so I think that static should also be removed

**Review by jdeppe-pivotal — APPROVED** [review]

**Pull request merged**